src/refdb_reftable.c

```diff
@@ -0,0 +1,514 @@
+{
+	reftable_backend *backend = GIT_CONTAINER_OF(_backend, reftable_backend, parent);
+	struct ref_record record;
+	int error;
+
+	if ((error = stack_read_ref(backend->stack, refname, &record)) < 0)
+		return error;
+	if (error > 0)
+		return 0;
+
+	ref_record_clear(&record);
+	return 1;
+}
+
+static int reftable_reflog__ensure_log(git_refdb_backend *_backend, const char *name)
+{
+	/* TODO: doesn't make any sense in the context of reftable? */
+	GIT_UNUSED(_backend);
+	GIT_UNUSED(name);
+	return 0;
+}
+
+static int reftable_reflog__read(git_reflog **out, git_refdb_backend *_backend, const char *name)
+{
+	reftable_backend *backend = GIT_CONTAINER_OF(_backend, reftable_backend, parent);
+	struct merged_table *table = NULL;
+	struct iterator iter = { 0 };
+	git_reflog *reflog;
+	int error;
+
+	reflog = git__calloc(1, sizeof(git_reflog));
+	GIT_ERROR_CHECK_ALLOC(reflog);
+
+	reflog->ref_name = git__strdup(name);
+	GIT_ERROR_CHECK_ALLOC(reflog->ref_name);
+
+	if (git_vector_init(&reflog->entries, 0, NULL) < 0)
+		goto out;
+
+	table = stack_merged_table(backend->stack);
+	GIT_ERROR_CHECK_ALLOC(table);
+
+	if ((error = merged_table_seek_log(table, &iter, "")) < 0)
+		goto out;
+
+	while (1) {
+		struct log_record record;
+		git_reflog_entry *entry;
+
+		if ((error = iterator_next_log(iter, &record)) < 0)
+			goto out;
+		if (error > 0)
+			break;
+
+		entry = git__calloc(1, sizeof(*entry));
+		GIT_ERROR_CHECK_ALLOC(entry);
+
+		if ((error = git_signature_new(&entry->committer,
+					       record.name, record.email,
+					       record.time, record.tz_offset)) < 0)
+			goto out;
+		entry->msg = record.message;
+		record.message = NULL;
+
+		if ((error = git_oid_fromraw(&entry->oid_old, record.old_hash)) < 0 ||
+		    (error = git_oid_fromraw(&entry->oid_cur, record.new_hash)) < 0)
+			goto out;
+
+		if ((git_vector_insert(&reflog->entries, entry)) < 0)
+			goto out;
+	}
+	error = 0;
+
+	*out = reflog;
+	reflog = NULL;
+out:
+	git_reflog_free(reflog);
+	iterator_destroy(&iter);
+	merged_table_free(table);
+	return error;
+}
+
+static int reftable_reflog__write(git_refdb_backend *backend, git_reflog *reflog)
+{
+	size_t entries = git_reflog_entrycount(reflog), i;
+	struct log_record *logs;
+	int error = 0;
+
+	logs = git__calloc(entries, sizeof(*logs));
+
+	for (i = 0; i < entries; i++) {
+		const git_reflog_entry *entry = git_reflog_entry_byindex(reflog, i);
+		if (entry == NULL) {
+			error = -1;
+			goto out;
+		}
+
+		if ((error = reftable_log_fill(&logs[i], entry->committer,
+					       &entry->oid_old, &entry->oid_cur,
+					       NULL, entry->msg)) < 0)
+			goto out;
+	}
+
+
+	if ((error = reftable_write_records(backend, NULL, 0, logs, entries)) < 0)
+		goto out;
+
+out:
+	git__free(logs);
+	return error;
+}
+
+static int reftable_reflog__rename(git_refdb_backend *_backend, const char *old_name, const char *new_name)
+{
+	GIT_UNUSED(_backend);
+	GIT_UNUSED(old_name);
+	GIT_UNUSED(new_name);
```

Comment:
Is there a better way to rename a reflog than to go through all reflog entries and adjust their name?

Reply:
The spec says


"Contrary to traditional reflog (which is a file), renames are encoded as a combination of ref deletion and ref creation."

but yes, you could delete and create entries too (but you need to compact the whole thing to get rid of traces of the data)

Reply:
Understood, thanks!